File: java/IEC61970/InfIEC61970/InfPart303/NetworkModelProjects/AnnotatedProjectDependency.java
package IEC61970.InfIEC61970.InfPart303.NetworkModelProjects;

import IEC61970.Base.Core.IdentifiedObject;

/**
 * @author selaost1
 * @version 1.0
 * @created 25-Dec-2023 8:31:54 PM
 */
public class AnnotatedProjectDependency extends IdentifiedObject {

	public DependencyType dependencyType;
	public NetworkModelProjectStage DependentOnStage;
	public NetworkModelProjectStage DependingStage;

	public AnnotatedProjectDependency(){

	}
}//end AnnotatedProjectDependency

File: py/IEC61970/InfIEC61970/InfPart303/NetworkModelProjects/AnnotatedProjectDependency.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Sat Jan  6 17:13:56 2024
# Class to represent AnnotatedProjectDependency
from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC61970.InfIEC61970.InfPart303.NetworkModelProjects.DependencyType import DependencyType
from IEC61970.InfIEC61970.InfPart303.NetworkModelProjects.NetworkModelProjectStage import NetworkModelProjectStage


class AnnotatedProjectDependency(IdentifiedObject):
    # constructor
    def __init__(self):
        super().__init__()
        self.dependency_type = DependencyType.REQUIRED  # Typical value for dependency type
        self.dependent_on_stage = NetworkModelProjectStage()  # Typical value for dependent on stage
        self.depending_stage = NetworkModelProjectStage()  # Typical value for depending stage
